${level} add format option (full, single char and ordinal)

c0shea opened the pull request:
Adds single character and ordinal level output options to the LevelLayoutRenderer to address #1949 and #1950

304NotModified commented:
cool! Thanks!

Could you place the `enum` in a new file and outside the class? Please check other enums for the naming style. 

to sync the csproj files, see https://github.com/NLog/NLog/blob/master/CONTRIBUTING.md#sync-projects

304NotModified commented on src/NLog/LayoutRenderers/LevelLayoutRenderer.cs:
```diff
@@ -46,13 +48,44 @@ namespace NLog.LayoutRenderers
     public class LevelLayoutRenderer : LayoutRenderer
     {
         /// <summary>
+        /// A value indicating when the cache is cleared.
```
I think this is wrong docs?

304NotModified commented on src/NLog/LayoutRenderers/LevelLayoutRenderer.cs:
```diff
@@ -46,13 +48,44 @@ namespace NLog.LayoutRenderers
+        [Flags]
+        public enum LevelOutputOption
+        {
+            /// <summary>Ouput the name as the level.</summary>
```
Proposal: 
- render the full level name
- render the first character for the level
- render the ordinal (aka number) for the level.

304NotModified commented on src/NLog/LayoutRenderers/LevelLayoutRenderer.cs:
```diff
@@ -46,13 +48,44 @@ namespace NLog.LayoutRenderers
+        /// Gets or sets a value indicating how the level is output.
+        /// </summary>
+        [DefaultValue(LevelOutputOption.Name)]
+        public LevelOutputOption LevelOutput { get; set; }
```
I was thinking of "format", what do you think?

304NotModified commented on tests/NLog.UnitTests/LayoutRenderers/LogLevelTests.cs:
```diff
@@ -60,5 +60,53 @@ public void LogLevelTest()
+            logger.Error("a");
+            AssertDebugLastMessage("debug", "E a");
+            logger.Fatal("a");
+            AssertDebugLastMessage("debug", "F a");
```
looks good! please add "trace"

304NotModified commented on tests/NLog.UnitTests/LayoutRenderers/LogLevelTests.cs:
```diff
@@ -60,5 +60,53 @@ public void LogLevelTest()
+            logger.Error("a");
+            AssertDebugLastMessage("debug", "4 a");
+            logger.Fatal("a");
+            AssertDebugLastMessage("debug", "5 a");
```
also please add trace

c0shea commented:
You're welcome :)

I like your suggestion to name it Format, so I renamed the enum to LevelFormat and moved it to its own file. I modeled it after StackTraceFormat, so hopefully it follows the style. I also removed the wrong doc and updated the unit tests to add trace.

Let me know if you need anything else to accept.

304NotModified commented:
Looks ready to merge! Could you please update the docs? Thanks!

304NotModified commented:
Fixes #1949 and fixes #1950

304NotModified commented:
docs see https://github.com/NLog/NLog/issues/2043

Will merge this an release a beta

c0shea commented:
Thanks! I have updated the docs with the new parameter and added a condensed table.

304NotModified commented:
FYI,

NLog 4.4.6 is live! 

https://www.nuget.org/packages/NLog/4.4.6
